lib/provider/repositories_provider.dart

```diff
@@ -4,20 +4,14 @@ import '../core/http/api_response.dart';
 import '../models/get_repositories/repository_response.dart';
 import '../repository/get_repositories_repo.dart';
 
-abstract class RepositoriesProvider {
```

Comment:
One thing I would like yoy to try.
- Add this abstract class back
- Make it implements Listenable.
- You may need to implement some other methods on RepositoriesProviderImpl.
- Where you are providing via ChangeNotifierProvider, change to ListenableProvider<RepositoriesProvider>

This way I think we can get back the interface, and use only it everywhere.

Reply:
done. Thanks for your help on this.